# HG changeset patch
# User Rik <rik@octave.org>
# Date 1399048807 25200
#      Fri May 02 09:40:07 2014 -0700
# Node ID a50d0d85dc73b913e2a92bfcd93dd8d1d7a3c66e
# Parent  89055521e04d7f22243c95f4ce7494c1766b1f50
Set axis equal for 2D parametric plots done with ezplot.

* __ezplot__.m: Use 'axis equal' for 2D parametric plots.

diff --git a/scripts/plot/draw/private/__ezplot__.m b/scripts/plot/draw/private/__ezplot__.m
--- a/scripts/plot/draw/private/__ezplot__.m
+++ b/scripts/plot/draw/private/__ezplot__.m
@@ -443,18 +443,22 @@ function [h, needusage] = __ezplot__ (pl
         if (i == 1)
           set (hax, "nextplot", "add");
         endif
       endfor
       set (hax, "nextplot", hold_state);
       axis (hax, domain);
     elseif (isplot || ispolar)
       h = feval (pltfunc, hax, X, Z);
-      if (isplot && ! parametric)
-        axis (hax, domain);
+      if (isplot)
+        if (! parametric)
+          axis (hax, domain);
+        else
+          axis ("equal");
+        endif
       endif
     elseif (isplot3)
       if (animate)
         ## draw animation, then replace with true plot3
         comet3 (hax, X, Y, Z, .05);
       endif
       h = feval (pltfunc, hax, X, Y, Z);
       set (hax, "box", "off");
